test/widgets/inbox_test.dart

```diff
@@ -385,6 +426,30 @@ void main() {
           checkAppearsUncollapsed(tester, 1, findSectionContent);
         });
 
+        testWidgets('collapse stream section after scroll', (tester) async {
+          await setupVarious(tester);
+
+          final listFinder = find.byType(Scrollable);
+          // Scroll the [StickyHeaderListView] enough so that
+          // the [_StreamSection] shows a sticky header
+          await tester.drag(listFinder, const Offset(0, -200));
+
+          final headerRow = findStreamHeaderRow(tester, 1);
+          // Check that the sticky header is present
+          check(headerRow).isNotNull();
+
+          final findSectionContent = find.text('specific topic');
+
+          checkAppearsUncollapsed(tester, 1, findSectionContent);
+          await tapCollapseIcon(tester, 1);
+          // Check that the header is still visible even after
+          // collapsing the section
+          check(headerRow).isNotNull();
+          checkAppearsCollapsed(tester, 1, findSectionContent);
```

Comment:
With the same cross-check as in my other comment — running the new test on the old code — this test also fails to cleanly fail, but a bit differently. Here's the output:
```
$ git checkout origin lib/
$ flutter test test/widgets/inbox_test.dart
[…]
00:03 +12 -1: InboxPage collapsing stream section collapse stream section after scroll                
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure was thrown running a test:
Expected: a Widget? that:
  is not null
Actual: <null>

When the exception was thrown, this was the stack:
#0      check.<anonymous closure> (package:checks/src/checks.dart:85:9)
#1      _TestContext.nest (package:checks/src/checks.dart:751:12)
#2      NullableChecks.isNotNull (package:checks/src/extensions/core.dart:128:20)
#3      main.<anonymous closure>.<anonymous closure>.<anonymous closure>.checkAppearsCollapsed (file:///home/greg/z/flutterz/test/widgets/inbox_test.dart:405:28)
#4      main.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (file:///home/greg/z/flutterz/test/widgets/inbox_test.dart:448:11)
<asynchronous suspension>
#5      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:183:15)
<asynchronous suspension>
#6      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:1018:5)
<asynchronous suspension>
<asynchronous suspension>
(elided one frame from package:stack_trace)

The test description was:
  collapse stream section after scroll
════════════════════════════════════════════════════════════════════════════════════════════════════
00:03 +12 -2: InboxPage collapsing stream section collapse stream section after scroll [E]            
  Test failed. See exception logs above.
  The test description was: collapse stream section after scroll
```

So it does fail, as one would hope — but on line 448, at `checkAppearsCollapsed`. This part:
```dart
          // Check that the header is still visible even after
          // collapsing the section
          check(headerRow).isNotNull();
```
again fails to detect that anything's wrong.